content/engineering/incidents/playbooks/ci.md

```diff
@@ -0,0 +1,226 @@
+    - Try [Restarting the agents](#restarting-the-agents) and restart the build.
+
+1. Is that flake related to the code:
+
+- See the process describe in the [flaky tests page](https://docs.sourcegraph.com/dev/background-information/testing_principles#flaky-tests)
+
+### Is this a failure or a flake?
+
+- Gravity: _minor_
+- Impact: Some builds will fail randomly, creating noise and slowing down the engineering team
+- Possible causes:
+  - Tests relying on timing.
+  - Race conditions.
+  - End to end tests are delicate by nature and can fail randomly due to the complexity of all involved components.
+  - State dependent test is not properly teared down and fails.
+
+#### Actions
+
+1. Immediately restart the faulty step.
+   - 💡 It will save you time while you're looking at the logs.
+   - Is the step passing now?
+     - **Yes**: See [Spotted a flake scenario](#spotted-a-flake)
+     - **No**: Give it another try, and see next point.
+1. Check on [Grafana](https://sourcegraph.grafana.net/explore?orgId=1&left=%5B%22now-12h%22,%22now%22,%22grafanacloud-sourcegraph-logs%22,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bapp%3D%5C%22buildkite%5C%22%7D%22%7D%5D) if there are any occurrences of the failures that were previously observed:
+1. Go the the "Explore" section
+1. Make sure to select `grafanacloud-sourcegraph-logs` in the dropdown at the top of page.
+1. Scope the time window to `7 Days` to make sure to find previous occurrences if there are any
+1. Enter a query such as `{app="buildkite"} |= "your error message"` where "your error message" is a string that identiy approximately the failure cause observed in the failing step.
+1. Is there a build that failed exactly like this?
+   - **Yes**:
+     1. 💡 Double check that you're looking at that the same step by inspecting the labels of message (click on the line to make them visible)
+     1. **Yes**, that's a flake. See the [Spotted a flake scenario](#spotted-a-flake)
+   - **No**: it's not a flake, reach out the team owning those tests.
+
+### Restarting the agents
+
+- Gravity: _minor_
+- Impact: May fail ongoing builds, but that's fine.
+- Possible causes:
+  - Manual restart by an engineer.
+  - Newer version of the agents needs to be deployed.
+
+#### Actions
+
+1. Open `k9s` to observe the currently running agents.
+1. In a different terminal, run `kubectl -n buildkite rollout restart deployment buildkite-agent`.
+1. Wait a bit to see the agents restarting completely.
+1. Restart the faulty build and observe it the problem is fixed or not.
+   - If necessary: escalate by creating an incident (`/incident` on Slack).
+
+### SSH'ing into an agent
+
+- Gravity: none
+- Impact: none (unless a destructive action is performed)
+- Possible cause:
+  - Need to investigate a problem and suspect the agent is at fault
+
+#### Actions
+
+1. SSH into an agent to reproduce the problem:
+1. Launch `k9s` and select an agent, they are named like `buildkite-agent-xxxxxxxxxx-yyyyy`
+1. Press `s` to SSH on it.
```

Comment:
Same as above, I would suggest `kubectl exec -it` here
```suggestion
1. Use `kubectl exec -it buildkite-agent-xxxx-yyyy` to open a shell 
```
Also, `kubectl exec` is not the same as SSH 😉 https://goteleport.com/blog/ssh-vs-kubectl/